llama_index/vector_stores/azurecosmosmongo.py

```diff
@@ -0,0 +1,250 @@
+"""Azure CosmosDB MongoDB vCore Vector store index.
+
+An index that that is built on top of an existing vector store.
+
+"""
+import logging
+import os
+from typing import Any, Dict, List, Optional, cast
+
+from pymongo import MongoClient
```

Comment:
could you move this into the the class definition, so that it's imported lazily? 

This is so that we don't have to add third party integrations as core dependencies in the main framework.

Reply:
That may not be possible as input typing depends on the import. mongodb_client: Optional[MongoClient] = None. Please note that similar pattern is used for mongodb: https://github.com/run-llama/llama_index/blob/main/llama_index/vector_stores/mongodb.py. However, if this is causing issues I can change the argument to "mongodb_client: Optional[any]" and do the lazy import. Please note that the tests should skip in absence of pymongo.

Reply:
@Disiok: please see ^ and let me know if you still prefer the lazy import?

Reply:
@Disiok: I applied the lazy import and also made some minor changes to other files to comply with linting, etc ..